pkg/test/framework/components/istio/config.go

```diff
@@ -166,6 +167,38 @@ func DefaultConfig(ctx resource.Context) (Config, error) {
 	return s, nil
 }
 
+func DefaultConfig2() (Config, error) {
```

Comment:
This is the main problem. While we allow test developer to customize, we can't construct from an empty `Config`, losing the cmd flag etc.

The obstacle to use `DefaultConfig()` directly is because 
- We setup the timeout duration differently based on `ctx ResourceTracker`, which is not really available when we call `SetupOnKube`. Get reference of that seems non trivial.
- `flag.Parse` is not invoked by in `TestMain() {SetupOnKube()}`

I hacked these two in `DefaulConfig2`.

Let me know what's the better way of doing these kinda of things. @ozevren

Reply:
SetupOnKube is a convenience method. You can implement one yourself where you have access to SuiteContext. That is where you can obtain the configuration.

Alternatively, we can refactor the istio.Config, so that it only contains user overridable things (or have a structure that allows "Defaulting" to config from commandline, if things are not explicitly specified.

Reply:
Thanks for explanation. I thought a little bit.

I think adding another package method along with SetupOnKube can confuse people.

Refactoring istio.Config or dynamically merging two structs seem overkill to me. I instead change `SetupOnKube` signature to accept a callback, which allow people to modify the configuration before we Deploy Istio.